pkg/client/client.go

```diff
@@ -352,6 +385,146 @@ func (c *serviceImpl) doRequest(method, path string, reqpl, respl any) error {
 	return nil
 }
 
+const openConnections = 10
+
+type filesUploader struct {
+	Client         *http.Client
+	EndpointURL    string
+	Filenames      []string
+	ChunkSizeBytes int64
+	DumpOut        io.Writer
+}
+
+type fileInfo struct {
+	Name        string
+	TotalChunks int
+}
+
+func (u *filesUploader) Upload() error {
+	var totalChunks int
+	var infos []fileInfo
+	for _, name := range u.Filenames {
+		stat, err := os.Stat(name)
+		if err != nil {
+			return nil
+		}
+		info := fileInfo{
+			Name:        name,
+			TotalChunks: int(math.Ceil(float64(stat.Size()) / float64(u.ChunkSizeBytes))),
+		}
+		infos = append(infos, info)
+		totalChunks += info.TotalChunks
+	}
+	jobsChan := make(chan *uploadChunkJob, openConnections)
+	resultsChan := make(chan error, 1)
+	defer close(jobsChan)
+	defer close(resultsChan)
+	for i := 0; i < openConnections; i++ {
+		w := &uploadChunkWorker{
+			Client:      u.Client,
+			EndpointURL: u.EndpointURL,
+			DumpOut:     u.DumpOut,
+			JobsChan:    jobsChan,
+			ResultsChan: resultsChan,
+		}
+		go w.Start()
+	}
```

Comment:
don't understand the comment

Reply:
Create a function named `startWorkers` and extract this section to it.

I thought I had left another comment about simplifying this function because it's too long but can't find it now so...: this function is too long and can be refactored into some calls to smaller functions (nit)

Reply:
Done